app/src/main/java/com/naman14/timberx/notifications/Notifications.kt

```diff
@@ -41,42 +41,42 @@ import com.naman14.timberx.ui.activities.MainActivity
 import com.naman14.timberx.util.Constants.ACTION_NEXT
 import com.naman14.timberx.util.Constants.ACTION_PLAY_PAUSE
 import com.naman14.timberx.util.Constants.ACTION_PREVIOUS
-import com.naman14.timberx.util.Utils.isOreo
+import com.naman14.timberx.util.Utils
+import com.naman14.timberx.util.doAsync
 import com.naman14.timberx.util.extensions.isPlaying
 import java.lang.System.currentTimeMillis
+import androidx.media.app.NotificationCompat as NotificationMediaCompat
 
 private const val CHANNEL_ID = "timberx_channel_01"
 private const val NOTIFICATION_ID = 888
 
-// TODO this should be an injected instance rather than a object singleton, use DI
-object NotificationUtils {
-    private var postTime: Long = 0
+interface Notifications {
 
-    private fun createNotificationChannel(context: Context) {
-        if (isOreo()) {
-            // TODO should be in strings.xml
-            val name = "Media playback"
-            val manager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
-            val channel = NotificationChannel(CHANNEL_ID, name, IMPORTANCE_LOW).apply {
-                // TODO should be in strings.xml
-                description = "Media playback controls"
-                setShowBadge(false)
-                lockscreenVisibility = Notification.VISIBILITY_PUBLIC
-            }
-            manager.createNotificationChannel(channel)
-        }
+    fun updateNotification(mediaSession: MediaSessionCompat)
+}
+
+class RealNotifications(
```

Comment:
We have both an interface and a "Real" implementation - this allows you to truly hide the actual implementation from consumers so they are only concerned about a surfaced API. This is common practice with DI 😄